### PR TITLE
Blazor security guidance updates

### LINKED ISSUE #17598: User.Identity.Name empty when implementing Azure B2C in Blazor WebAssembly
This issue is about the following doc page:
[Secure an ASP.NET Core Blazor WebAssembly hosted app with Azure Active Directory B2C](https://docs.microsoft.com/en-us/aspnet/core/security/blazor/webassembly/hosted-with-azure-active-directory-b2c?view=aspnetcore-3.1)

## Description
When trying to access *HttpContext.User.Identity.Name* in the Server project inside an API that is protected by the [Authorize] attribute, the name will appear empty. This while I am logged in at the name is shown in the Client application just fine.

## Fix
I had to add the following snippet in Startup.cs to fix this.

```C#
services.AddAuthentication(AzureADB2CDefaults.JwtBearerAuthenticationScheme)
		.AddAzureADB2CBearer(options =&gt; Configuration.Bind("AzureAd", options));

services.Configure&lt;JwtBearerOptions&gt;(AzureADB2CDefaults.JwtBearerAuthenticationScheme, options =&gt;
{
	options.TokenValidationParameters = new Microsoft.IdentityModel.Tokens.TokenValidationParameters()
	{
		NameClaimType = "name"
	};
});
```

I am not sure if this is the right place to report it, not even sure if it is something that should be fixed. At most it will hopefully help other people that come across the same issue and save them some time.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7d682d4c-86b5-c47e-89a3-3684b2235320
* Version Independent ID: a80e115c-466a-288d-a918-9ef7514f5add
* Content: [Secure an ASP.NET Core Blazor WebAssembly hosted app with Azure Active Directory B2C](https://docs.microsoft.com/en-us/aspnet/core/security/blazor/webassembly/hosted-with-azure-active-directory-b2c?view=aspnetcore-3.1#feedback)
* Content Source: [aspnetcore/security/blazor/webassembly/hosted-with-azure-active-directory-b2c.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/security/blazor/webassembly/hosted-with-azure-active-directory-b2c.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #17598

@IIStevowII ... see how this looks to you.

### CODE REVIEW

**Comment by IIStevowII** [comment]
Looks good.

There is one thing that I just noticed, which might be interesting as well.
In the docs, the AddAuthentication() differs from my own implementation.

### Docs
```csharp
services
    .AddAuthentication(AzureADDefaults.BearerAuthenticationScheme)
    .AddAzureADBearer(options => Configuration.Bind("AzureAd", options));
```
### My implementation
```csharp
services
    .AddAuthentication(AzureADB2CDefaults.JwtBearerAuthenticationScheme)
    .AddAzureADB2CBearer(options => Configuration.Bind("AzureAd", options));
```

It seems to be a completely different package, `Microsoft.AspNetCore.Authentication.AzureAD.UI` vs `Microsoft.AspNetCore.Authentication.AzureADB2C.UI`

I'd expect that the B2C implementation would use the ADB2C version.

**Comment by guardrex** [comment]
I think you're correct ... it's do to an INCLUDE. I'll fix that right now. Thanks for catching it.

The docs were all correct, then I got the bright idea to roll in some INCLUDES files to reduce duplicated content. *That's when 😈 got into the works!*

**Comment by guardrex** [comment]
Actually ... it turns out that it's just a simple cut-'n-paste error on my part. There's no INCLUDE for that one. I did screw up an INCLUDE earlier. 🙈

**Comment by guardrex** [comment]
kk ... I think that licked the main problems here for now. Thank goodness it's all still in preview and there's time left to iron out the rough spots. Thanks for your help, @IIStevowII!

I'll merge and take this live immediately.

File: aspnetcore/security/blazor/webassembly/hosted-with-azure-active-directory-b2c.md [hunk 1]
```diff
@@ -111,20 +111,20 @@ To specify the output location, which creates a project folder if it doesn't exi
...
 ```xml
-<PackageReference Include="Microsoft.AspNetCore.Authentication.AzureAD.UI" 
+<PackageReference Include="Microsoft.AspNetCore.Authentication.AzureADB2C.UI" 
     Version="3.1.0" />
 ```
 
 ### Authentication service support
 
-The `AddAuthentication` method sets up authentication services within the app and configures the JWT Bearer handler as the default authentication method. The `AddAzureADBearer` method sets up the specific parameters in the JWT Bearer handler required to validate tokens emitted by the Azure Active Directory:
+The `AddAuthentication` method sets up authentication services within the app and configures the JWT Bearer handler as the default authentication method. The `AddAzureADBearer` method sets up the specific parameters in the JWT Bearer handler required to validate tokens emitted by the Azure Active Directory B2C:
```
Comment (IIStevowII) [inline aspnetcore/security/blazor/webassembly/hosted-with-azure-active-directory-b2c.md:123]: AddAzureADBearer should be AddAzureADB2CBearer as well.

Comment (guardrex) [reply aspnetcore/security/blazor/webassembly/hosted-with-azure-active-directory-b2c.md:123]: **_Roasted Rat Burgers!_** 🐀🍔 ..... I'll fix it. (*Sneaky Code!*:tm:).

**Pull request merged**